js/customize-snapshots.js

```diff
@@ -731,16 +701,69 @@
 		 * @returns {string} URL.
 		 */
 		getSnapshotFrontendPreviewUrl: function getSnapshotFrontendPreviewUrl() {
-			var snapshot = this, a = document.createElement( 'a' );
+			var snapshot = this, a = document.createElement( 'a' ),
+				params = {};
+
+			if ( api.settings.changeset && api.settings.changeset.uuid ) {
+				params.customize_changeset_uuid = api.settings.changeset.uuid;
+			} else {
+				params.customize_changeset_uuid = snapshot.data.uuid;
+			}
+
 			a.href = snapshot.frontendPreviewUrl.get();
-			if ( a.search ) {
-				a.search += '&';
+			if ( snapshot.statusButton.disbleButton.get() ) {
+				return a.href;
+			}
+			if ( ! api.settings.theme.active ) {
+				params.theme = api.settings.theme.stylesheet;
 			}
-			a.search += snapshot.uuidParam + '=' + snapshot.data.uuid;
+			a.search = $.param( params );
+
 			return a.href;
 		},
 
 		/**
+		 * Frontend preview link setup.
+		 *
+		 * @returns {void}
+		 */
+		setUpPreviewLink: function setUpPreviewLink() {
+			var snapshot = this;
+
+			snapshot.previewLink = $( $.trim( wp.template( 'snapshot-preview-link' )() ) );
+			snapshot.previewLink.attr( 'target', snapshot.data.uuid );
+			snapshot.previewLink.href = snapshot.frontendPreviewUrl.get();
+
+			snapshot.previewLink.click( function( e ) {
+				var onceProcessingComplete;
+				e.preventDefault();
+
+				snapshot.previewLink.href = snapshot.getSnapshotFrontendPreviewUrl();
```

Comment:
@westonruter Added the same logic for updating the href as is for opening the link, do you think that's fine? (See the latest commit, `f50139b`).